src/functions.c

```diff
@@ -549,10 +550,64 @@ static void fcallCommandGeneric(client *c, int ro) {
         return;
     }
 
+    if ((fi->f_flags & FUNCTION_FLAG_NO_CLUSTER) && server.cluster_enabled) {
+        addReplyError(c, "Can not run function on cluster, 'no-cluster' flag is set.");
+        return;
+    }
+
+    if ((fi->f_flags & FUNCTION_FLAG_DENY_OOM) && server.script_oom) {
```

Comment:
There's the question whether we want to make this also dependent on `FUNCTION_FLAG_WRITE`. Intuitively we want to deny running when OOM only for `write` functions.
Here are the most likely scenarios and what I think should be the relevant flags:
1. A read only function, can run even under OOM: no flags are set.
2. A write function, we want to avoid it running under OOM: ideally we just mark it as `write` and implicitly get `deny-oom` as well. 
3. A write function which we want to explicitly allow under OOM conditions: ideally we mark it as `write` and as `allow-oom`.

Reply:
@oranagra and I talked about it and the conclusion was that if the user does un-natural things like deny-oom on readonly command then we should not care. If we will go with your suggestion it will not align with CMD_* flags that already exists today for commands. @oranagra WDYT?

Reply:
I think `CMD_DENYOOM` is an internal flag and isn't user facing so we don't need to consider being "similar" to it. Also I'd change it in a separate PR to `CMD_ALLOW_OOM_WRITE` but I know what you're going to say about the blame log... On the other hand since this is generated from the new _commands.json_ this might be a good time to do the change...

Reply:
it's not internal, it's exposed to modules and even clients (via COMMAND command)

Reply:
ok